pyoozie/workflow.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2017 "Shopify inc." All rights reserved.
+# Use of this source code is governed by a MIT-style license that can be found in the LICENSE file.
+import typing  # pylint: disable=unused-import
+
+from pyoozie import tags  # pylint: disable=unused-import
+
+
+class AbstractAction(object):
+    pass
+
+
+class Action(AbstractAction):
+    """Concrete action to execute (takes an action tag, e.g. pyoozie.Shell, pyoozie.Email. etc.)"""
+
+    def __init__(self, action_tag, on_error=None):
```

Comment:
I'm not sure if `None` is an appropriate default, but I'm also not sure what to use instead. Maybe some kind of `Defer`? How do you see error handlers composing? Or are we just going to walk up the graph until we find one?